garden/ros2_gazebo_integration_tutorial.md

```diff
@@ -0,0 +1,113 @@
+The `robot_state_publisher` takes the description and joint angles of the robot as inputs and publishes the 3D poses of the robot links, using a kinematic tree model of the robot. 
+
+```python
+    robot_state_publisher = Node(
+        package='robot_state_publisher',
+        executable='robot_state_publisher',
+        name='robot_state_publisher',
+        output='both',
+        parameters=[
+            {'use_sim_time': True},
+            {'robot_description': robot_desc},
+        ]
+    )
+```
+
+These 3D poses are published on the `/tf` topic which is useful to track/plan robot's motion, communicate between different parts of the robot or evaluate robot's performance.
+
+### Configure a communication bridge
+
+These joint states can either come from `joint_state_publisher` as seen earlier or from simulated [JointStatePub](https://gazebosim.org/api/gazebo/4.5/classignition_1_1gazebo_1_1systems_1_1JointStatePublisher.html#details) system's joint_state message.
+
+Configure a bridge between ROS topic `/joint_states` and Gazebo topic `/world/demo/model/diff_drive/joint_state` by adding remappings in the node setup or by creating a bridge.yaml: 
+
+```bash
+- ros_topic_name: "/joint_states"
+  gz_topic_name: "/world/demo/model/diff_drive/joint_state"
+  ros_type_name: "sensor_msgs/msg/JointState"
+  gz_type_name: "gz.msgs.Model"
+  direction: GZ_TO_ROS
+```
+
+Learn more about the bridge from [ROS 2 Integration](docs/garden/ros2_integration).
+
+### Maintaining a single robot description format
+
+Main pain point of using existing simulation assets with ROS 2 tools is updating URDF files into Gazebo readable format, out of scope for this tutorial, is no longer required. If you are maintaining a URDF and an SDF file in a project, you can now drop the URDF and just use SDF for both ROS and Gazebo.  
+This is made possible by `sdformat_urdf`, parser plugin library that converts an SDF file to URDF C++ DOM structures, making it understandle by the ROS 2 ecosystem. 
+Though there are some limitations of the plugin like not all SDFormat tags are compatible, for example if you have any sensors attached to a joint, it won't be parsed. More details [here](https://github.com/ros/sdformat_urdf/tree/ros2/sdformat_urdf).
+
+To embed this functionality, we simply need to print the SDFormat file to `/robot_description` ROS topic and internally it'll find a suitable parser, `sdformat_urdf` in this case, to read the file. This is already done while configuring the `robot_state_publisher` earlier.
+
+### Run RViz and Gazebo
+
+The second aim defined in [Setup](#Setup) is essentially maintaining only one robot description format which now can be controlled with directly with Gazebo. Take SDFormat XML describing a robot and publish joint states from the simulator, letting `robot_state_publisher` handle turning those joint states into tf transforms which is used by RViz and other ROS tools.
+
+![gz_rviz](tutorials/ros2_integration/gz_rviz.mp4)
```

Comment:
I don't know if the mp4 files will play :thinking: I haven't seen movie files embedded in these tutorials. Will need to test to find out.